components/match2/wikis/halo/match_summary.lua

```diff
@@ -102,10 +102,26 @@ function CustomMatchSummary.getByMatchId(args)
 		match.opponents[2].type == Opponent.team
 	then
 		local team1, team2 = string.gsub(match.opponents[1].name, ' ', '_'), string.gsub(match.opponents[2].name, ' ', '_')
-		match.links.headtohead = tostring(mw.uri.fullUrl('Special:RunQuery/Head2head')) ..
-		'?pfRunQueryFormName=Match+history&Head_to_head_query%5Bplayer%5D=' .. team1 ..
-		'&Head_to_head_query%5Bopponent%5D=' .. team2 .. '&wpRunQuery=Run+query'
+		local buildQueryFormLink = function(form, template, arguments)
+			return tostring(mw.uri.fullUrl('Special:RunQuery/' .. form,
+				mw.uri.buildQueryString(Table.map(arguments, function(key, value) return template .. key, value end))
```

Comment:
Why does every key need to be prefixed with 'h2h'? (genuine question)

Reply:
This is just the way MW queries function, in the form page this would be the "for template" for input fields.

Reply:
See also https://www.mediawiki.org/wiki/Extension:Page_Forms/Creating_query_forms#Preloading_data_in_the_query

Reply:
So... For the Halo Wiki, The Form is called "Head2head", which uses the template "Headtohead", which calls the module "Head2HeadOverview" (Not on Git)

So, the "Headtohead" before every term is setting (injecting?) the template argument, then the value for that argument. From what I can tell. I'm not 100% sure how forms work.

Reply:
The form should be the same as the one made for Rainbow Six but specifically adapted to Halo by @DMeluca